### ISSUE TITLE
scoop uninstall blocked when app is running

### ISSUE THREAD

**Issue by deevus** [lukesampson/scoop#488]
Example:

```
$ scoop uninstall openssh
uninstalling openssh (5.4p1-1)
removing shim for scp
removing shim for sftp
removing shim for slogin
removing shim for ssh-add
removing shim for ssh-agent
removing shim for ssh-keygen
removing shim for ssh-keyscan
removing shim for ssh
couldn't remove ~/appdata/local/scoop\apps\openssh\5.4p1-1: it may be in use
```

I'm thinking that users should be able to uninstall apps regardless of whether it is running or not - which means that we need some way of stopping services before uninstalling, so that the user doesn't need to figure it out themselves.

**Comment by jonasgeiregat** [comment]
Agreed! But then there should be a way to add services in the first place not ?

**Comment by Ash258** [comment]
First thing to do when uninstalling should be check for directory is in use. 

I could think only using `Get-Process`. Something like: 

```powershell
if ((Get-Process -FileVersionInfo -ErrorAction SilentlyContinue | ? { $_.FileName -like '$dir`*'}).Count -gt 0) {
     abort 'Folder is in use. Make sure to close program'
}
```

https://github.com/lukesampson/scoop/blob/99b5afafa086246aac65d63160218c38b014650d/libexec/scoop-uninstall.ps1#L41-L46

$dir should be evaluated before writing information about uninstalling (L43)

@r15ch13 Do you have better solution?

**Comment by Ash258** [comment]
Result of get-process for runing Steam

![Steam](https://i.imgur.com/8IyoZis.png)

```text
PS C:\Windows\system32> (Get-Process -FileVersionInfo -ErrorAction SilentlyContinue | ? { $_.FileName -like 'D:\scoop\ap
ps\Steam*'})

ProductVersion   FileVersion      FileName
--------------   -----------      --------
2018.2.13.864... 2018.2.13.864... D:\SCOOP\apps\Steam\current\steamapps\common\God Awe-full Clicker\GAC.exe
03.00.00.01      04.83.53.91      D:\SCOOP\apps\Steam\current\GameOverlayUI.exe
01.00.00.01      04.83.53.91      D:\SCOOP\apps\Steam\current\Steam.exe
01.00.00.01      04.83.53.91      D:\SCOOP\apps\Steam\nightly-20181111\bin\cef\cef.win7\steamwebhelper.exe
01.00.00.01      04.83.53.91      D:\SCOOP\apps\Steam\nightly-20181111\bin\cef\cef.win7\steamwebhelper.exe
01.00.00.01      04.83.53.91      D:\SCOOP\apps\Steam\nightly-20181111\bin\cef\cef.win7\steamwebhelper.exe
01.00.00.01      04.83.53.91      D:\SCOOP\apps\Steam\nightly-20181111\bin\cef\cef.win7\steamwebhelper.exe
01.00.00.01      04.83.53.91      D:\SCOOP\apps\Steam\nightly-20181111\bin\cef\cef.win7\steamwebhelper.exe
01.00.00.01      04.83.53.91      D:\SCOOP\apps\Steam\nightly-20181111\bin\cef\cef.win7\steamwebhelper.exe
                                  D:\SCOOP\apps\Steam\current\steamapps\common\God Awe-full Clicker\UnityCrashHandle...


PS C:\Windows\system32> (Get-Process -FileVersionInfo -ErrorAction SilentlyContinue | ? { $_.FileName -like 'D:\scoop\ap
ps\Steam*'}).Count
10
```

**Comment by r15ch13** [comment]
Good idea 👍


It's a bit faster when the `-FileVersionInfo` isn't used. We don't need the file version for this. The `Path` property is enough.
Warning the user and do nothing would be the best way. This should also be implemented in the update process to skip running programs.

```powershell
(Get-Process | ? { $_.Path -like 'D:\scoop\apps\*'} )
(Get-Process | ? { $_.Path -like 'D:\scoop\apps\<app>\*'} )
```

```powershell
λ (Get-Process | ? { $_.Path -like 'D:\scoop\apps\*'} )

Handles  NPM(K)    PM(K)      WS(K)     CPU(s)     Id  SI ProcessName
-------  ------    -----      -----     ------     --  -- -----------
   2238     176   322672     451496     746,63   3832   8 firefox
   1654     198   868320     820768     701,23  17776   8 firefox
   1506     150   454816     436532     107,84  19904   8 firefox
    823      33   153628     170696     387,61  20420   8 firefox
    809     138   390228     475888      79,34  27380   8 firefox
   1524     166   481256     425892     300,42  30668   8 firefox
   1544     165   553188     494080     229,72  30920   8 firefox
    216      15     4828      83404       0,16  32460   8 firefox
    448      25    53392      58904       1,31  34272   8 QuickLook
    126       9     1320       6180       0,03  18816   8 QuickLook.WoW64HookHelper
    355      29    44316      68388       0,92   7316   8 Wavebox
    328      32    56140      82212      28,56  13692   8 Wavebox
    551      49    87400     117192       2,64  17148   8 Wavebox
    361      53   154552     172872      28,86  20368   8 Wavebox
   1646      83   128120     166200      74,45  20576   8 Wavebox
    738      43   369040     146876      21,42  20764   8 Wavebox
    305      24    26568      35984       0,11  22416   8 Wavebox
    370      75   252996     286688      52,16  24092   8 Wavebox
    357      43    88072     117712      32,67  34092   8 Wavebox
    211      14     6776      12616       0,03  34792   8 Wavebox


λ (Get-Process | ? { $_.Path -like 'D:\scoop\apps\wavebox\*'} )

Handles  NPM(K)    PM(K)      WS(K)     CPU(s)     Id  SI ProcessName
-------  ------    -----      -----     ------     --  -- -----------
    355      29    44316      68400       0,92   7316   8 Wavebox
    325      32    56116      82576      28,70  13692   8 Wavebox
    551      49    87400     117428       2,64  17148   8 Wavebox
    361      53   154552     172944      28,86  20368   8 Wavebox
   1661      85   128088     165180      74,70  20576   8 Wavebox
    738      43   369040     146876      21,42  20764   8 Wavebox
    302      24    26144      36320       0,11  22416   8 Wavebox
    379      76   246384     281988      52,16  24092   8 Wavebox
    359      43    87176     117444      32,81  34092   8 Wavebox
    211      14     6776      12616       0,03  34792   8 Wavebox
```

**Comment by Ash258** [comment]
Fixed as soon as #3608 arrive in master

Proper implementation will follow in #2952

**Comment by rasa** [comment]
Implemented in [https://github.com/lukesampson/scoop/pull/3608](https://github.com/lukesampson/scoop/pull/3608)
